### PR TITLE
Added AWS Cognito & Jira Notifications

### PR DESCRIPTION
**Pull request by danmenzies-jerram**
* Removed unnecessary Cognito error checking
* Added SSL (needed for oAuth 2.0 authentication via Cognito)
* Added new config options for cognito
* Added relevant node packages
* Adjusted the `test-server` script to allow for SSL
* Added detailed instructions on setting up Cognito

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -24,6 +24,19 @@ const {
 
 const config = require('./config');
 
+configSsl = config.get("ssl");
+if (configSsl && configSsl.certificate_path) {
+    var https = require('https');
```
Comment (DesignyourCode) [inline src/server.js]: `const`

File: src/server.js [hunk 2]
```diff
@@ -24,6 +24,19 @@ const {
...
+    var fs = require('fs');
```
Comment (DesignyourCode) [inline src/server.js]: `const`

File: src/server.js [hunk 3]
```diff
@@ -24,6 +24,19 @@ const {
...
+
+    var sslOptions = {
```
Comment (DesignyourCode) [inline src/server.js]: `const`

File: src/server.js [hunk 4]
```diff
@@ -168,8 +181,16 @@ function run(db) {
         db.run('INSERT OR IGNORE INTO user (id,name,blocked,trusted,created_at) VALUES (1,"dev",0,1,datetime())');
     }
 
-    var server = app.listen(config.get('port'), config.get('host'), (err) => {
-        if (err) throw err;
-        console.log(`server listening on ${server.address().port}`);
-    });
+    if (configSsl && configSsl.certificate_path) {
+        server = https.createServer( sslOptions, app );
```
Comment (DesignyourCode) [inline src/server.js]: Something doesn't seem quite right with the server var being created.

Either we need to use a let above the if statement `let server;`

And then remove the `var` on the server below.

Or we need to make this `const server` on this line, and then change the `var server` below to be `const server` also.

**Review by DesignyourCode — CHANGES_REQUESTED** [review]

File: src/server.js (hunk 1 above)
Comment (danmenzies-jerram) [reply src/server.js]: Fixed

File: src/server.js (hunk 2 above)
Comment (danmenzies-jerram) [reply src/server.js]: Also fixed

File: src/server.js (hunk 3 above)
Comment (danmenzies-jerram) [reply src/server.js]: Fixed

File: src/server.js (hunk 4 above)
Comment (danmenzies-jerram) [reply src/server.js]: I've changed this to `let`; as the `if` statement is required; If the next Dev removes the certificate, or if this is deployed across an insecure connection, the server needs the ability to serve over `http` - which uses a completely different library to the secured connection over `https`.

**Review by DesignyourCode — APPROVED** [review]

**Pull request merged**